sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -204,12 +206,17 @@ object Cast {
       > SELECT _FUNC_('10' as int);
        10
   """)
-case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String] = None)
+case class Cast(
+    child: Expression,
+    dataType: DataType,
+    timeZoneId: Option[String] = None,
+    format: Option[String] = None)
```

Comment:
Based on this PR, it sounds like format is only applicable to datetime columns. Could you show us the full spec of this syntax? Is it applicable to numeric columns?

This is a useful feature, but any mainstream SQL system supports this syntax?

Reply:
You can find **6.13 \<cast specification\>** at https://www.iso.org/standard/63556.html.
Applies only for **datetime data type -> character string data type** and **character string type -> datetime data type casts**.

Oracle DB already supports a similar formatting option during [cast](https://docs.oracle.com/en/database/oracle/oracle-database/18/sqlrf/CAST.html)
but `FORMAT` is omitted, and there is an initiative in Impala: [IMPALA-4018](https://issues.apache.org/jira/browse/IMPALA-4018) and Hive [HIVE-21575](https://issues.apache.org/jira/browse/HIVE-21575) to support.

Reply:
I think we should hold it until the other mainstream database support it. Basically, no system supports this new format yet.

Reply:
Fine. I fixed the UT failures anyway.